pkg/lib/files/files_test.go

```diff
@@ -239,12 +259,89 @@ func TestListDirRecursive(t *testing.T) {
 	require.NoError(t, err)
 	require.ElementsMatch(t, expected, filesListRecursive)
 
-	filesListRecursive, err = ListDirRecursive(tmpDir, true, IgnoreNonPython)
+	filesListRecursive, err = ListDirRecursive(tmpDir, true, emptyExcludes, IgnoreNonPython)
 	expected = []string{
 		filepath.Join("2.py"),
 		filepath.Join("3/1.py"),
 		filepath.Join("3/2/1.py"),
 	}
 	require.NoError(t, err)
 	require.ElementsMatch(t, expected, filesListRecursive)
+
+	filesListRecursive, err = ListDirRecursive(tmpDir, false, excludes)
+	expected = []string{
+		filepath.Join(tmpDir, "2.py"),
+		filepath.Join(tmpDir, "3/2/2.txt"),
+		filepath.Join(tmpDir, "4/2.pyc"),
+		filepath.Join(tmpDir, "4/.git/HEAD"),
+	}
+	require.NoError(t, err)
+	require.ElementsMatch(t, expected, filesListRecursive)
+
+	filesListRecursive, err = ListDirRecursive(tmpDir, true, excludesWithRelativePaths, IgnoreNonPython)
+	expected = []string{
+		filepath.Join("2.py"),
+	}
+	require.NoError(t, err)
+	require.ElementsMatch(t, expected, filesListRecursive)
+}
+
+func TestReadAllIgnorePatterns(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "cortexignore-test")
+	if err != nil {
```

Comment:
`require.NoError(t, err)` (here and below) might be a little cleaner, but not a big deal either way, so feel free to make the change or not based on your preference

Reply:
Agree. It looks cleaner.